applications/StructuralMechanicsApplication/custom_elements/prestress_membrane_element.cpp

```diff
@@ -1132,142 +1135,552 @@ void PrestressMembraneElement::CalculateMembraneElasticityTensor(
+            CrossProduct(mG3Initial[PointNumber],mG1Initial[PointNumber],mG2Initial[PointNumber]);
+            mG3Initial[PointNumber] /= norm_2(mG3Initial[PointNumber]);    
+            
+        }
+
+        // project prestress in the plane spanned by G1 and G2 and transform it in the cartesian CoSy in reference configuration
+        // -- in case of isotropic prestress: no transformation necessary (isotropic state is the same in every (cartesian) CoSy)
+        // -- in case of anisotropic prestress: projection is only executed in the first step, in the following steps the prestress is updated 
+        //    based on the current state
+        if(mAnisotropicPrestress && rCurrentProcessInfo[TIME] == 1){
+            // Initialize Prestress
+            for (unsigned int i_strain=0; i_strain<strain_size; i_strain++){
+                prestress_variable(i_strain,PointNumber) = GetProperties()[MEMBRANE_PRESTRESS](0,i_strain);
+                if(GetProperties().Has(PRESTRESS_AXIS_2_GLOBAL) && GetProperties().Has(PRESTRESS_AXIS_1_GLOBAL) == true){
+                    prestress_axis_1(i_strain,PointNumber) = GetProperties()[PRESTRESS_AXIS_1_GLOBAL](0,i_strain);
+                    prestress_axis_2(i_strain,PointNumber) = GetProperties()[PRESTRESS_AXIS_2_GLOBAL](0,i_strain);
+                }
+            } 
+            // in case that no prestress directions are prescribed: hardcode the prestress direction
+            if (GetProperties().Has(PRESTRESS_AXIS_2_GLOBAL) && GetProperties().Has(PRESTRESS_AXIS_1_GLOBAL) == false){
+                prestress_axis_1(0,PointNumber) = 1;
+                prestress_axis_1(1,PointNumber) = 0;
+                prestress_axis_1(2,PointNumber) = 0;
+            
+                prestress_axis_2(0,PointNumber) = 0;
+                prestress_axis_2(1,PointNumber) = 1;
+                prestress_axis_2(2,PointNumber) = 0;
+            }
+        
+            TransformPrestress(PointNumber);
+        }
+
+        // in case of isotropic prestress: set prestress (no transformation necessary)
+        if(mAnisotropicPrestress == false){
+            for (unsigned int i_strain=0; i_strain<strain_size; i_strain++){
+                prestress_variable(i_strain,PointNumber) = GetProperties()[MEMBRANE_PRESTRESS](0,i_strain);
+            }
+        }
+    }
+
+    // Initialize Material
+
+    if (mConstitutiveLawVector.size() == 0)
+    {
+        mConstitutiveLawVector.resize(integration_points.size());
+
+        for (unsigned int i = 0; i < mConstitutiveLawVector.size(); i++)
+        {
+            mConstitutiveLawVector[i] = GetProperties()[CONSTITUTIVE_LAW]->Clone();
+
+            mConstitutiveLawVector[i]->InitializeMaterial(GetProperties(), GetGeometry(), row(GetGeometry().ShapeFunctionsValues(), i));
+        }
+    }
+}
+void PrestressMembraneElement::InitializeNonLinearIteration(){
+    
+}
+void PrestressMembraneElement::UpdatePrestress(unsigned int PointNumber){
```

Comment:
This function is almost 300 lines long!
This makes it very hard to maintain/refactor/debug
Could you please split it up into several smaller functions?
You already have some comment abt how you could separate them

Reply:
done